etl/data_helpers/geo.py

```diff
@@ -323,13 +323,24 @@ def add_region_aggregates(
     df_region = pd.DataFrame({country_col: [], year_col: []}).astype(dtype={country_col: "object", year_col: "int"})
     # Select data for countries in the region.
     df_countries = df[df[country_col].isin(countries_in_region)]
+    # Add population for doing weighted average aggregations
+    df_countries = add_population_to_dataframe(df_countries)
```

Comment:
I was checking why tests are failing and adding population to a dataframe is the reason. It might be better to add extra argument `weight` to the function and call it like this

```python
def add_data_for_regions(tb: Table, regions: List[str], ds_regions: Dataset, ds_income_groups: Dataset) -> Table:
    tb_with_regions = tb.copy()
    aggregations = {column: "mean" for column in tb_with_regions.columns if column not in ["country", "year"]}
    tb_with_regions = add_population_to_dataframe(tb_with_regions)

    for region in REGIONS:
        # Find members of current region.
        members = geo.list_members_of_region(
            region=region,
            ds_regions=ds_regions,
            ds_income_groups=ds_income_groups,
        )
        tb_with_regions = geo.add_region_aggregates(
            df=tb_with_regions,
            region=region,
            countries_in_region=members,
            countries_that_must_have_data=[],
            num_allowed_nans_per_year=None,
            frac_allowed_nans_per_year=0.2,
            aggregations=aggregations,
            weights=tb_with_regions['population']
        )
```

This is more explicit and reusable than adding population.

Then the function becomes
```python
def weighted_mean(x, w):
    return np.ma.average(
        np.ma.masked_invalid(x),
        weights=np.ma.masked_invalid(w),
    )

if weights is not None:
    assert aggregations[variable] == "mean", "Weights only work mean aggregation"
    variable_agg = lambda x: weighted_mean(x, weights.loc[x.index])
```

Reply:
@Marigold  thanks a lot! it seems like my unit tests are still failing because of using a lambda function here, but more annoyingly I am not sure how to best resolve an issue where I can't get a World aggregate here, because population is added before regions that the World aggregate depends on. I actually remember having this issue before which is why I moved the population step addition inside the add_regions function. Do you by any chance have an idea for a quick fix for this? No worries if not ofc! I'll keep trying to fix it in the mean time

Reply:
Oh right, so if your previous approach worked well, perhaps it'd be best to copy `add_region_aggregates` to your step and rewrite it to calculate the weighted mean (while adding population).

(Another approach would be running `add_region_aggregates` first for `population * grade`, then for `population` and divide them to get the weighted mean.)

Reply:
perfect, thanks! I think this is the easiest solution for now